File: android/walletlink/src/main/java/com/coinbase/walletlink/dtos/JoinSessionMessageDTO.kt
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

package com.coinbase.walletlink.dtos

import com.coinbase.wallet.core.interfaces.JsonSerializable
import com.coinbase.wallet.core.util.JSON
import com.coinbase.walletlink.models.ClientMessageType

/**
 * Client message to join currently active WalletLink session
 *
 * @property type Type of message
 * @property id Client generated request ID
 * @property sessionId Server generated session ID
 * @property sessionKey Client computed session key
 */
internal data class JoinSessionMessageDTO(
    val type: ClientMessageType = ClientMessageType.JoinSession,
    val id: Int,
    val sessionId: String,
    val sessionKey: String
) : JsonSerializable {
    @ExperimentalUnsignedTypes
    override fun asJsonString(): String = JSON.toJsonString(this)
}

File: ios/Source/DTOs/JoinSessionMessageDTO.swift
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

import CBCore

/// Client message to join currently active WalletLink session
struct JoinSessionMessageDTO: Codable, JSONDeserializable, JSONSerializable {
    /// Type of message
    let type: ClientMessageType = .joinSession

    /// Client generated request ID
    let id: Int32

    /// Server generated session ID
    let sessionId: String

    /// Client computed session key
    let sessionKey: String
}
